### PR TITLE
Feature/ecbuild35

### PR DESCRIPTION
**Pull request by mmiesch**
**Description**

This is part of a series of PRs for a description see:

https://github.com/JCSDA-internal/fv3-bundle/pull/50

This introduces minor updates for compatibility with ecbuild 3.5.0

partially fixes: https://github.com/JCSDA-internal/jedi-stack/issues/19

**How Has This Been Tested?**

This has been incorporated into JEDI fv3-bundle and all tests have passed

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [ ] `make distcheck` passes

### CODE REVIEW

**Review by danholdaway — APPROVED** [review]

**Review by ytremolet — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,9 +48,13 @@ else()
   ecbuild_enable_ompstubs()
 endif()
 
+# jedi-cmake
+find_package( jedicmake QUIET)
+
 # NetCDF
+set( NETCDF_F90 ON CACHE BOOL "Compile with Fortran NetCDF" )
```
Comment (markjolah) [inline CMakeLists.txt:55]: This variable is not used by jedi cmake FindNetCDF.

File: CMakeLists.txt [hunk 2]
```diff
@@ -53,9 +48,13 @@ else()
...
 find_package( NetCDF REQUIRED COMPONENTS Fortran )
-include_directories( ${NETCDF_INCLUDE_DIR} )
+include_directories( ${NetCDF_INCLUDE_DIRS} )
```
Comment (markjolah) [inline CMakeLists.txt:57]: `include_directories()` is not necessary for NetCDF targets.

File: CMakeLists.txt [hunk 3]
```diff
@@ -331,10 +330,12 @@ list( APPEND fms_src_files
 ################################################################################
 ecbuild_add_library( TARGET   fms
                      SOURCES  ${fms_hdr_files} ${fms_src_files}
-                     LIBS     ${NETCDF_LIBRARIES}
+                     PUBLIC_LIBS     ${NetCDF_LIBRARIES}
```
Comment (markjolah) [inline CMakeLists.txt:333]: This should be `PUBLIC_LIBS NetCDF::NetCDF_Fortran`

File: CMakeLists.txt [hunk 4]
```diff
@@ -331,10 +330,12 @@ list( APPEND fms_src_files
...
+                     PUBLIC_INCLUDES ${NetCDF_INCLUDE_DIRS}
```
Comment (markjolah) [inline CMakeLists.txt:334]: This is not necessary.  The NetCDF targets are already exporting their include directories.

**Review by markjolah — CHANGES_REQUESTED** [review]
See the documentation for [jedicmake/FindNetCDF](https://github.com/JCSDA-internal/jedi-cmake/blob/develop/cmake/Modules/FindNetCDF.cmake).  This should explain how to properly use the Module.  There is no need to use `include_directories()`.  The jedicmake NetCDF targets already export all the paths and libraries you need.  All that is needed to `target_link_libraries(<tgt> PUBLIC NetCDF::NetCDF_<LANG>)`.  This is what is happening when you set the `PUBLIC_LIBS NetCDF::NetCDF_<LANG>` argument to `ecbuild_add_library()`.

File: CMakeLists.txt [hunk 5]
```diff
@@ -343,19 +344,23 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES GNU AND CMAKE_Fortran_COMPILER_VERSION VERS
...
+#set(CMAKE_REQUIRED_DEFINITIONS "-D_GNU_SOURCE=1")
+#check_symbol_exists("gettid" "unistd.h" HAVE_GETTID)
+#unset(CMAKE_REQUIRED_DEFINITIONS)
+#if(HAVE_GETTID)
+#    target_compile_definitions(fms PRIVATE HAVE_GETTID)
+#endif()
 
 if(ECBUILD_INSTALL_FORTRAN_MODULES)
-  install(DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/${CMAKE_CFG_INTDIR} DESTINATION ${INSTALL_INCLUDE_DIR} )
+   install(DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/${CMAKE_CFG_INTDIR} DESTINATION ${INSTALL_INCLUDE_DIR} )
```
Comment (markjolah) [inline CMakeLists.txt:356]: Change `INSTALL_INCLUDE_DIR` -> `CMAKE_INSTALL_INCLUDEDIR`.

File: CMakeLists.txt [hunk 6]
```diff
@@ -343,19 +344,23 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES GNU AND CMAKE_Fortran_COMPILER_VERSION VERS
...
 endif()
 
+target_include_directories( fms INTERFACE
+                    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>/include
+                    $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
+                    $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}> )
```
Comment (markjolah) [inline CMakeLists.txt:362]: Change `INSTALL_INCLUDE_DIR` -> `CMAKE_INSTALL_INCLUDEDIR`.

**Comment by mmiesch** [comment]
@markjolah - when I make your requested changes, the build fails with this:

```
[ 34%] Linking Fortran shared library ../lib/libfms.dylib
Undefined symbols for architecture x86_64:
  "___netcdf_MOD_nf90_close", referenced from:
      ___netcdf_io_mod_MOD_netcdf_file_close in netcdf_io.F90.o
  "___netcdf_MOD_nf90_copy_att", referenced from:
      ___blackboxio_MOD_copy_metadata in blackboxio.F90.o
  "___netcdf_MOD_nf90_create", referenced from:
      ___blackboxio_MOD_create_diskless_netcdf_file in blackboxio.F90.o
      ___netcdf_io_mod_MOD_netcdf_file_open in netcdf_io.F90.o
  "___netcdf_MOD_nf90_def_dim", referenced from:
      ___blackboxio_MOD_copy_metadata in blackboxio.F90.o
      ___netcdf_io_mod_MOD_netcdf_add_dimension in netcdf_io.F90.o
  "___netcdf_MOD_nf90_def_var_manydims", referenced from:
      ___blackboxio_MOD_copy_metadata in blackboxio.F90.o
      ___netcdf_io_mod_MOD_netcdf_add_variable in netcdf_io.F90.o
  "___netcdf_MOD_nf90_def_var_scalar", referenced from:
      ___netcdf_io_mod_MOD_netcdf_add_variable in netcdf_io.F90.o
  "___netcdf_MOD_nf90_enddef", referenced from:
      ___netcdf_io_mod_MOD_set_netcdf_mode in netcdf_io.F90.o
  "___netcdf_MOD_nf90_get_att_eightbyteint", referenced from:
      ___netcdf_io_mod_MOD_get_variable_attribute_1d in netcdf_io.F90.o
      ___netcdf_io_mod_MOD_get_global_attribute_1d in netcdf_io.F90.o
  "___netcdf_MOD_nf90_get_att_eightbytereal", referenced from:
      ___netcdf_io_mod_MOD_get_variable_attribute_1d in netcdf_io.F90.o

[...]
```

File: CMakeLists.txt (hunk 3 above)
Comment (mmiesch) [reply CMakeLists.txt:333]: It looks like this is the culprit.  If I make this change, the build fails.  The other changes seem to work.

**Comment by markjolah** [comment]
@mmiesch You need to build with `VERBOSE=1` to see the actual compilation lines.  It looks like you are not correctly getting the FindNetCDF from `jedicmake`, and perhaps still using that provided from `ecbuild`.

**Comment by mmiesch** [comment]
Thanks @markjolah but I don't want to spend too much time debugging this - I've got other things to debug with test failures and containers.  The point of these PRs was just to get things working, and they work.  We can refine & clean up the cmake later.

**Review by markjolah — APPROVED** [review]

**Pull request merged**